hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManagerStarter.java

```diff
@@ -164,7 +166,15 @@ private void commonInit() {
     public void start(OzoneConfiguration conf) throws Exception {
       StorageContainerManager stm = StorageContainerManager.createSCM(conf);
       stm.start();
-      stm.join();
+
+      ShutdownHookManager.get().addShutdownHook(() -> {
```

Comment:
should we add a longer timeout here? The default 1s may be too short.

Reply:
I see the default is 30 seconds.

```
  /** Default shutdown hook timeout: {@value} seconds. */
  public static final long SERVICE_SHUTDOWN_TIMEOUT_DEFAULT = 30;

  static long getShutdownTimeout(Configuration conf) {
    long duration = conf.getTimeDuration(
        SERVICE_SHUTDOWN_TIMEOUT,
        SERVICE_SHUTDOWN_TIMEOUT_DEFAULT,
        TIME_UNIT_DEFAULT);
    if (duration < TIMEOUT_MINIMUM) {
      duration = TIMEOUT_MINIMUM;
    }
    return duration;
  }
```

Reply:
Introduced a config and handled this as part of #2301